contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,805 @@
+
+        // check if can be jailed
+        // there are two slots for each day
+        uint256 dayIndex1 = (block.timestamp / 1 days) * 2;
+        uint256 dayIndex2 = dayIndex1 + 1;
+        if (_canBeJailed(dayIndex1)) {
+            _setJailed(dayIndex1);
+        } else if (_canBeJailed(dayIndex2)) {
+            _setJailed(dayIndex2);
+        } else {
+            revert("NO_JAIL_SLOT");
+        }
+
+        // slash
+        (bool canSlash, uint256 jailUntil) = _checkSlashRecord(operatorAddress, SlashType.DoubleSign);
+        require(canSlash, "ALREADY_SLASHED");
+        uint256 slashAmount = IStakeCredit(valInfo.creditContract).slash(doubleSignSlashAmount);
+        _jailValidator(valInfo, jailUntil);
+
+        emit ValidatorSlashed(operatorAddress, jailUntil, slashAmount, SlashType.DoubleSign);
+
+        _sync(valInfo.creditContract, operatorAddress);
+    }
+
+    /*----------------- gov -----------------*/
+    function pauseStaking() external onlyAssetProtector {
+        _stakingPaused = true;
+        emit StakingPaused();
+    }
+
+    function resumeStaking() external onlyAssetProtector {
+        _stakingPaused = false;
+        emit StakingResumed();
+    }
+
+    function addBlackList(address _addr) external onlyAssetProtector {
+        blackList[_addr] = true;
+    }
+
+    function removeBlackList(address _addr) external onlyAssetProtector {
+        blackList[_addr] = false;
+    }
+
+    function updateParam(string calldata key, bytes calldata value) external onlyGov {
+        if (key.compareStrings("transferGasLimit")) {
+            require(value.length == 32, "INVALID_VALUE_LENGTH");
+            uint256 newTransferGasLimit = value.bytesToUint256(32);
+            require(newTransferGasLimit >= 2300, "INVALID_TRANSFER_GAS_LIMIT");
+            transferGasLimit = newTransferGasLimit;
+        } else if (key.compareStrings("minSelfDelegationBNB")) {
+            require(value.length == 32, "INVALID_VALUE_LENGTH");
+            uint256 newMinSelfDelegationBNB = value.bytesToUint256(32);
+            require(newMinSelfDelegationBNB >= 1000 ether, "INVALID_MIN_SELF_DELEGATION_BNB");
+            minSelfDelegationBNB = newMinSelfDelegationBNB;
+        } else if (key.compareStrings("minDelegationBNBChange")) {
+            require(value.length == 32, "INVALID_VALUE_LENGTH");
+            uint256 newMinDelegationBNBChange = value.bytesToUint256(32);
+            require(newMinDelegationBNBChange >= 1 ether, "INVALID_MIN_DELEGATION_BNB_CHANGE");
+            minDelegationBNBChange = newMinDelegationBNBChange;
+        } else if (key.compareStrings("maxElectedValidators")) {
+            require(value.length == 32, "INVALID_VALUE_LENGTH");
+            uint256 newMaxElectedValidators = value.bytesToUint256(32);
+            require(newMaxElectedValidators >= 1, "INVALID_MAX_ELECTED_VALIDATORS");
+            require(newMaxElectedValidators <= 500, "INVALID_MAX_ELECTED_VALIDATORS");
```

Comment:
use 500 to replace MAX_NUM_OF_VALIDATORS is more meaningful

Reply:
I think if the value is not as expected, we should let the transaction fail instead of changing it to a value different from the input